### PR TITLE
Design and implement CLI for sending data source related transaction

### LINKED ISSUE #525: Design and implement CLI for sending data source related transactions

### PR DESCRIPTION
**Pull request by taobun**
fixed #525 

Let's review this pr and used it as a reference for other commands.

### CODE REVIEW

File: chain/x/zoracle/client/cli/tx.go [hunk 1]
```diff
@@ -2,18 +2,28 @@ package cli
...
+	"github.com/cosmos/cosmos-sdk/version"
 	"github.com/cosmos/cosmos-sdk/x/auth"
 	"github.com/cosmos/cosmos-sdk/x/auth/client/utils"
 	"github.com/spf13/cobra"
 )
 
+const (
+	flagName    = "name"
+	flagScript  = "script"
+	flagCallFee = "call-fee"
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go:24]: Let's call this `flagDataSourceFee = "dsfee"`

File: chain/x/zoracle/client/cli/tx.go [hunk 2]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
 		},
 	}
 }
+
+// GetCmdCreateDataSource implements the create data source command handler
+func GetCmdCreateDataSource(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-data-source (--name [name]) (--script [path_to_script]) (--call-fee [fee]) (--owner [owner])",
+		Short: "Create new data source",
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: a new

File: chain/x/zoracle/client/cli/tx.go [hunk 3]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
 		},
 	}
 }
+
+// GetCmdCreateDataSource implements the create data source command handler
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: fullstop

File: chain/x/zoracle/client/cli/tx.go [hunk 4]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
...
+	}
+	cmd.Flags().String(flagName, "", "name of data source")
+	cmd.Flags().String(flagScript, "", "path to data source script")
+	cmd.Flags().String(flagCallFee, "", "fee for query this data source")
+	cmd.Flags().String(flagOwner, "", "owner of this data source")
+
+	return cmd
+}
+
+// GetCmdEditDataSource implements the edit data source command handler
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: also here

File: chain/x/zoracle/client/cli/tx.go [hunk 5]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
...
+
+			name, err := cmd.Flags().GetString(flagName)
+			if err != nil {
+				return err
+			}
+			scriptPath, err := cmd.Flags().GetString(flagScript)
+			if err != nil {
+				return err
+			}
+			execBytes, err := readFile(scriptPath)
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Can we use `ioutil`? https://golang.org/pkg/io/ioutil

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go [hunk 6]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
...
+}
+
+// GetCmdEditDataSource implements the edit data source command handler
+func GetCmdEditDataSource(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "edit-data-source [id] (--name [name]) (--script [path_to_script]) (--call-fee [fee]) (--owner [owner])",
+		Short: "Edit data source",
+		Args:  cobra.ExactArgs(1),
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Edit existed data source by current owner.
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: Edit an existing data source. The caller must be the current data source's owner.

File: chain/x/zoracle/client/cli/tx.go [hunk 7]
```diff
@@ -118,3 +130,157 @@ func GetCmdReport(cdc *codec.Codec) *cobra.Command {
...
 }
+
+// GetCmdCreateDataSource implements the create data source command handler
+func GetCmdCreateDataSource(cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-data-source (--name [name]) (--script [path_to_script]) (--call-fee [fee]) (--owner [owner])",
+		Short: "Create new data source",
+		Args:  cobra.NoArgs,
+		Long: strings.TrimSpace(
+			fmt.Sprintf(`Create new data source that will be used by oracle script.
```
Comment (sorawit) [inline chain/x/zoracle/client/cli/tx.go]: by oracle scripts.

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go (hunk 1 above)
Comment (sorawit) [reply chain/x/zoracle/client/cli/tx.go:24]: Actually call fee is also ok. But I'd prefer the flag name be "callfee"

File: chain/x/zoracle/client/cli/tx.go (hunk 3 above)
Comment (sorawit) [reply chain/x/zoracle/client/cli/tx.go]: Great job writing the comment though!

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/x/zoracle/client/cli/tx.go (hunk 1 above)
Comment (sorawit) [reply chain/x/zoracle/client/cli/tx.go:24]: Ok let's go with "call-fee"

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**